src/main/java/servlets/UploadReceiptServlet.java

```diff
@@ -75,69 +78,58 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
    */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Optional<Entity> receipt = createReceiptEntity(request);
-
-    // Respond with status code 400, Bad Request.
-    if (!receipt.isPresent()) {
-      logger.warning("Valid JPEG file was not uploaded.");
-      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "No JPEG file uploaded.");
+    Entity receipt = null;
+
+    try {
+      receipt = createReceiptEntity(request);
+    } catch (FileNotSelectedException | InvalidFileException e) {
+      logger.warning(e.toString());
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      response.getWriter().println(e.toString());
+      return;
```

Comment:
I thought we were gonna show a message to the user here too, or was that somewhere else?

Reply:
The code for showing the message is in upload.js (L41)